src/_gettsim/taxes/zu_verst_eink/vorsorgeaufw.py

```diff
@@ -85,63 +114,90 @@ def _vorsorge_alternative_tu_ab_2005_bis_2009(  # noqa: PLR0913
     return out
 
 
+@dates_active(start="2005-01-01", end="2019-12-31", change_name="vorsorgeaufw_tu")
 @add_rounding_spec(params_key="eink_st_abzuege")
-def vorsorgeaufw_tu_ab_2005_bis_2009(
-    _vorsorge_alternative_tu_ab_2005_bis_2009: float,
+def vorsorgeaufw_tu_guenstiger(
     vorsorgeaufw_tu_bis_2004: float,
+    vorsorgeaufw_tu_einführung: float,
 ) -> float:
-    """Calculate Vorsorgeaufwendungen from 2005 to 2009.
+    """Calculate Vorsorgeaufwendungen from 2005 to 2019.
 
     With the 2005 reform, no taxpayer was supposed to be affected negatively.
     Therefore, one needs to compute amounts under the 2004 and 2005 regimes
     and take the more favourable one.
 
+    After a supreme court ruling, the 2005 rule had to be changed in 2010.
+    Therefore, one needs to compute amounts under the 2004 and 2010 regimes
+    and take the more favourable one. (§10 (3a) EStG).
+
+    Sidenote: The 2010 rules are by construction at least as beneficial as
+    the 2005 regime, so there is no need for a separate check.
+
     Parameters
     ----------
-    _vorsorge_alternative_tu_ab_2005_bis_2009
-        See :func:`_vorsorge_alternative_tu_ab_2005_bis_2009`.
+    vorsorgeaufw_tu
+        See :func:`vorsorgeaufw_tu`.
     vorsorgeaufw_tu_bis_2004
         See :func:`vorsorgeaufw_tu_bis_2004`.
 
     Returns
     -------
 
     """
-    out = max(vorsorgeaufw_tu_bis_2004, _vorsorge_alternative_tu_ab_2005_bis_2009)
+    out = max(vorsorgeaufw_tu_bis_2004, vorsorgeaufw_tu_einführung)
 
     return out
 
 
-@add_rounding_spec(params_key="eink_st_abzuege")
-def vorsorgeaufw_tu_ab_2010_bis_2019(
-    vorsorgeaufw_tu_bis_2004: float, vorsorgeaufw_tu_ab_2020: float
+@dates_active(
+    start="2010-01-01",
+    end="2019-12-31",
+    change_name="vorsorgeaufw_tu_einführung",
+)
+def _vorsorgeaufw_tu_ab_2020(  # noqa: PLR0913
```

Comment:
function names seems misleading

Reply:
Yeah, not totally happy with that module. But if we called it "..._2010" one would have to check that it is the actually the same as the version without an underscore.